invenio_vocabularies/datastreams/writers.py

```diff
@@ -110,3 +111,21 @@ def write(self, stream_entry, *args, **kwargs):
             yaml.safe_dump([stream_entry.entry], file)
 
         return stream_entry
+
+
+class AsyncWriter(BaseWriter):
```

Comment:
From what I understand, this approach is to use this writer as a container for other writer configs.

I was thinking if it would be possible to have a `delay` kwarg in `BaseWriter`, which if `True` would achieve the same result... But from what I see that's pretty difficult to handle, given `__init__` overrides, etc. 

My fear is that we end up with a very complex datastream config structure, where especially in YAML it's easy to make indentation mistakes, which as a result would eventually lead to more difficult UX/DX.

Other approaches to discuss:

- Move this up to the DataStream, i.e. in `BaseDataStream.write(...)`, do something like `if writer.delay: ...call task...`
- Preserve the original writer config/init-params when initializing writers?

Reply:
I agree with you. However, changes are not trivial.

In both cases we would need to implement a way to either save the configuration (_type_ + _args_) or make writers JSON serializable. Otherwise, we won't be able to send them to the task. IMO this fits more at data stream level, i.e. _the writers writes_, but is the data stream who decides _how/when_.